### ISSUE TITLE
[BUG] v1.27.0 does not save results of json report

### ISSUE THREAD

**Issue by radekBednarik** [microsoft/playwright#17953]
**Context:**
- Playwright Version: 1.27.0
- Operating System: Windows 10, Ubuntu 22.04 Docker Image
- Node.js version: LTS
- Browser: All
- Extra: this error started happeing after update to 1.27.0. Downgrade back to 1.26.1 makes this feature to work again. 

**Describe the bug**

Version 1.27.0 does not save `json` report data, even if `preserveOutput: "always",` is provided in config file.
It will only create the `test-results` folder, but it is empty.
Downgrade to 1.26.1 "solves" this bug.

My base config:

```ts
/**
 * This is the base config file, which is to be imported into the specific environments configurations files.
 * It can be then extended or overwritten, as needed.
 */

import { devices } from "@playwright/test";
import type { PlaywrightTestConfig } from "@playwright/test";
import dotenv from "dotenv";
import path from "path";

dotenv.config({ path: path.resolve(process.cwd(), "configs", "base.env") });

const baseConfig: PlaywrightTestConfig = {
  // run files and test() in spec.ts files in parallel
  fullyParallel: true,
  // base timeout
  timeout: 60000,
  // assertion timeout
  expect: {
    timeout: 5000,
  },
  // test directory root
  testDir: "../tests/e2e",
  // paralelization settings - 2 on CI pipeline, default on localhost
  workers: process.env.CI ? 2 : undefined,
  // no. of retries - must be at least one to trigger artifact creation opts below
  retries: 1,
  // reporters
  reporter: !process.env.CI
    ? [["line"], ["json", { outputFile: "test-results/results.json" }]]
    : [["dot"], ["json", { outputFile: "test-results/results.json" }]],
  // global setup script
  globalSetup: "../helpers/pwGlobalSetup.ts",
  // preserve output of test reporting/artefacts
  preserveOutput: "always",
  // global configuration for browser and browser context
  use: {
    headless: true,
    viewport: { width: 1920, height: 1080 },
    ignoreHTTPSErrors: true,
    video: "on-first-retry",
    trace: "on-first-retry",
    screenshot: "only-on-failure",
    actionTimeout: 30000,
    navigationTimeout: 30000,
    bypassCSP: true,
  },
  // test configurations
  projects: [
    {
      name: "desktop chrome",
      use: { ...devices["Desktop Chrome"] },
    },
    // TODO: disabled due to issue on K8S Ubuntu 22.04 docker agent
    // bug reported to PW team: https://github.com/microsoft/playwright/issues/17875
    // {
    //   name: "desktop firefox",
    //   use: { ...devices["Desktop Firefox"] },
    // },
    {
      name: "desktop safari",
      use: { ...devices["Desktop Safari"] },
    },
  ],
};

export { baseConfig };
```

**Comment by pavelfeldman** [comment]
> ["json", { outputFile: "test-results/results.json" }]]

As of 1.27, we are consistently resolving all the path inside the playwright.config relative to the config. Due to an oversight, this particular option was resolved relative to cwd previously. This has been addressed in 1.27.

**Comment by radekBednarik** [comment]
@pavelfeldman can confirm, adjusting the value of `outputFile` prop with resolving logic in mind got it working. Thank you. 
Suggestion: I can not recall, that the information of path resolution of `playwright.config` is somewhere in docs/guides. If that is the case, I would like to suggest to add this info somewhere.
